File: java/src/main/java/com/pbdviewer/utils/pbclass/PbVariableFlag.java
package com.pbdviewer.utils.pbclass;

public enum PbVariableFlag {
    IsCustom(1),
    IsShared(2),
    UShort(4),
    ULong(8),
    Invalid(12),
    IsBuffer(16),
    IsArray(32),
    IsPrivate(64),
    IsProtected(128);

    public final int value;

    PbVariableFlag(int value) {
        this.value = value;
    }

    public static boolean hasFlag(int flags, PbVariableFlag flag) {
        return (flags & flag.value) == flag.value;
    }
}

File: html/js/pbclass/PbVariableFlag.js
// PbVariableFlag.js — port of Java PbVariableFlag.java (enum)
// Values taken directly from the Java source:
//   IsCustom(1), IsShared(2), UShort(4), ULong(8), Invalid(12),
//   IsBuffer(16), IsArray(32), IsPrivate(64), IsProtected(128)
export const PbVariableFlag = Object.freeze({
    IsCustom:    1,
    IsShared:    2,
    UShort:      4,
    ULong:       8,
    Invalid:     12,
    IsBuffer:    16,
    IsArray:     32,
    IsPrivate:   64,
    IsProtected: 128,

    /**
     * @param {number} value  — the flags byte
     * @param {number} flag   — one of the constants above
     * @returns {boolean}
     */
    hasFlag(value, flag) {
        return (value & flag) === flag;
    },
});
